customer-models/src/main/java/org/powertac/customer/evcharger/DemandSampler.java

```diff
@@ -0,0 +1,183 @@
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.powertac.common.config.Configurator;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Sampler class which holds the logic of sampling new EV charger demand from
+ * the statistical model.
+ *
+ * @author Philipp Page <github@philipp-page.de>
+ */
+class DemandSampler
+{
+  private static final Logger log =
+    LogManager.getLogger(DemandSampler.class.getSimpleName());
+  private MixtureMultivariateNormalDistribution pluginProbability;
+  private Map<String, MixtureMultivariateNormalDistribution> condHorizonDemandProbabilities =
+    new HashMap<>();
+  private XMLConfiguration config;
+
+  void initialize ()
+  {
+    // TODO: Make this dynamic
+    String path = "config/residential_ev.xml";
+    try {
+      config = Configurator.readXML(path);
+    }
+    catch (ConfigurationException e) {
+      log.error("Problem loading configuration: " + e);
+    }
+    catch (Exception e) {
+      log.error("Error loading configuration: " + e);
+    }
+
+    setupPluginProbability();
+    setupDemandHorizonProbabilities();
+  }
+
+  // This will parse the XML config into a
+  // MixtureMultivariateNormalDistribution.
+  private void setupPluginProbability ()
+  {
+    double[] means = config.get(double[].class, "pluginProbability.means.mean");
+    double[] variances =
+      config.get(double[].class, "pluginProbability.covs.cov");
+    Array2DRowRealMatrix covariances = new Array2DRowRealMatrix(variances);
+    double[] weights = config.get(double[].class, "pluginProbability.weights");
+    List<Pair<Double, MultivariateNormalDistribution>> mvns =
+      new LinkedList<>();
+    for (int i = 0; i < means.length; i++) {
+      mvns.add(Pair
+              .create(weights[i],
+                      new MultivariateNormalDistribution(new double[] { means[i] },
+                                                         new double[][] { covariances
+                                                                 .getRow(i) })));
+    }
+    pluginProbability = new MixtureMultivariateNormalDistribution(mvns);
+  }
+
+  // This will parse the XML config into a
+  // HashMap<hour of day, MixtureMultivariateNormalDistribtion>.
+  // The demandHorizonProbabilities represent the conditional Gaussian Mixture
+  // Model.
+  private void setupDemandHorizonProbabilities ()
+  {
+    String[] instances = config.getStringArray("instances");
+
+    for (String instance: instances) {
+      final double[] flatMeans =
+        config.get(double[].class, String.format("%s.means.mean", instance));
+      final double[] flatCovs =
+        config.get(double[].class, String.format("%s.covs.cov", instance));
+      final double[] weights =
+        config.get(double[].class, String.format("%s.weights", instance));
+
+      final double[][] meanVectors = new double[weights.length][2];
```

Comment:
2 is the dimension of the model (bivariate normal distribution returning horizon, energy demand tuples).